docs/src/cookbooks/cookbook.md

```diff
@@ -66,6 +66,28 @@ class Foo extends RawModule {
 }
 ```
 
+### How can I tieoff a Bundle/Vec to 0?
+
+You can use `asTypeOf` as above, but also use `chiselTypeOf` rather than having to
+figure out the type of the LHS:
+
+```scala mdoc:silent:reset
+import chisel3._
+
+class MyBundle extends Bundle {
+  val foo = UInt(4.W)
+  val bar = Vec(4, UInt(1.W))
+}
+
+class Foo(gen: () => Data) extends RawModule {
```

Comment:
```suggestion
class Foo(gen: Data) extends RawModule {
```

Reply:
We probably shouldn't change this here, but is there a good reason why we name parameters like this `gen`?
Something like `outputType: Data` seems much more accurate.

Reply:
I guess because I always think of it as the "generator", since before you had to do something like:

```IO(Output(new FooBundle())```

I think of anything as a `new` as a generator.
Is this `Chisel._` thinking, not `chisel3` thinking?

Reply:
I think of it as the result of doing `chiselTypeOf(...)` or as a directly defined type `UInt(8.w)`

Reply:
I think `gen` came from older style where it would usually be a 0-arity "generator method", eg.
```scala
class MyBundle[T <: Data](gen: () => T) extends Bundle {
  val foo = gen()
  val bar = gen()
}
```
It still probably makes sense in those contexts, but I agree that in cases like this something like `outputType` makes more sense.